diff -urNp -x '*.orig' motif-2.3.8.org/ac_find_xft.m4 motif-2.3.8/ac_find_xft.m4
--- motif-2.3.8.org/ac_find_xft.m4	2017-03-28 00:24:31.000000000 +0200
+++ motif-2.3.8/ac_find_xft.m4	2023-12-08 22:45:43.438774053 +0100
@@ -75,7 +75,12 @@ saved_LIBS="$LIBS"
 LIBS="$LIBS $freetype_lib"
 saved_CPPFLAGS="$CPPFLAGS"
 CPPFLAGS="$CPPFLAGS $FREETYPE_CFLAGS"
-AC_CHECK_HEADERS(freetype/freetype.h)
+AC_CHECK_HEADERS(ft2build.h \
+		    freetype/freetype.h \
+		    freetype/ftglyph.h \
+		    freetype/tttables.h \
+		    freetype/ftnames.h \
+		    )
 
 FINDXFT_HAVE_FREETYPE="no"
 case "$ac_cv_header_freetype_freetype_h" in
