diff -urNp -x '*.orig' motif-2.3.8.org/configure.ac motif-2.3.8/configure.ac
--- motif-2.3.8.org/configure.ac	2023-12-08 22:45:00.669224607 +0100
+++ motif-2.3.8/configure.ac	2023-12-08 22:45:00.874227241 +0100
@@ -40,9 +40,9 @@ AM_PROG_LIBTOOL
 
 AC_EXEEXT
 
-dnl AC_PROG_YACC
+AC_PROG_YACC
 dnl Do this the old fashioned way. 'bison -y' doesn't cut it
-AC_CHECK_PROGS(YACC, byacc, yacc)
+dnl AC_CHECK_PROGS(YACC, byacc, yacc)
 
 dnl AC_PROG_SPLINT
 AC_CHECK_PROGS(SPLINT, splint)
